src/rgw/rgw_sal_motr.cc

```diff
@@ -2230,12 +2231,12 @@ int MotrObject::delete_obj_aio(const DoutPrefixProvider* dpp, RGWObjState* astat
 
 int MotrCopyObj_CB::handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len)
 {
-  progress_cb progress_CB = this->get_progress_cb();
   int rc = 0;
   ldpp_dout(m_dpp, 20) << "Offset=" << bl_ofs << " Length = "
                        << " Write Offset=" << write_offset << bl_len << dendl;
 
 
+  struct req_state* s = static_cast<req_state*>(obj_ctx->get_private());
```

Comment:
We should make `struct req_state* s` as a member variable instead of `obj_ctx`. Pass `obj_ctx` as reference in the  `MotrCopyObj_CB` constructor and initialize `struct req_state* s` in the constructor.

Reply:
@sjain09 not in all cases we would need req_state, so adding it as member variable in MotrCopyObj_CB would be an overhead for cases where its not being used. Secondly, req_state is already been maintained RGWObjectCtx, we are not adding it newly, but just retrieving for use here.

> class RGWObjectCtx {
  rgw::sal::Store* store;
  ceph::shared_mutex lock = ceph::make_shared_mutex("RGWObjectCtx");
  void *s{nullptr};

  std::map<rgw_obj, RGWObjState> objs_state;
public:
  explicit RGWObjectCtx(rgw::sal::Store* _store) : store(_store) {}
  explicit RGWObjectCtx(rgw::sal::Store* _store, void *_s) : store(_store), s(_s) {}

  void *get_private() {
    return s;
  }

Reply:
Here we are just adding one pointer variable in `MotrCopyObj_CB` I don't see any overhead there. We are going to use it repeatedly in the handle_data() function so it makes sense to initialize it only once.

Reply:
understood, thanks